src/connect/components/node-input/node-input.tsx

```diff
@@ -0,0 +1,69 @@
+import { useLayoutEffect, useRef, useState } from 'react';
+import { twMerge } from 'tailwind-merge';
+import { useEventListener, useOnClickOutside } from 'usehooks-ts';
+
+type Props = {
+    name?: string;
+    defaultValue?: string;
+    placeholder?: string;
+    'aria-label'?: string;
+    className?: string;
+    minLength?: number;
+    onSubmit: (value: string) => void;
+    onCancel: () => void;
+};
+export function NodeInput(props: Props) {
```

Comment:
This is a replacement for the existing `TreeItemInput` which had a dependency on react-aria. It was causing problems for me so I figured I might as well just get rid of it, since we were about to do that anyway.